src/Views/HardwareView.vala

```diff
@@ -554,6 +568,51 @@ public class About.HardwareView : Gtk.Box {
         }
     }
 
+    private void request_hostname_permission () {
+        if (hostname_permission != null) {
+            return;
+        }
+
+        try {
+            // Asks for permission to execute SetStaticHostname and SetPrettyHostname
+            hostname_permission = new Polkit.Permission.sync (
```

Comment:
We need to make sure we're setting both the static hostname and pretty hostname:

https://github.com/elementary/initial-setup/blob/c4108e028384c40ac31c5b0022e9224c1c314060/src/Views/AccountView.vala#L351